File: backend-node/internal/settings/lazy_config.js
const path = require("path");
const fs = require("fs");

const getFEPublicDir = () => {
  return path.resolve(__dirname, "../../../frontend/public");
};

const getTestDir = () => {
  return path.resolve(__dirname, "../../../tests");
};

const readConfigJson = () => {
  const configPath = path.join(getFEPublicDir(), "DBConfig.json");
  const config = JSON.parse(fs.readFileSync(configPath, "utf8"));
  return config;
};

module.exports = {
  getTestDir,
  readConfigJson,
};

File: backend-go/internal/settings/LazyConfig.go
package settings

import (
	"fmt"
	"io/ioutil"
	"os"
	"path/filepath"
)

func getFEPublicDir() string {
	cwd, err := os.Getwd()
	if err != nil {
		panic(err)
	}
	fp := filepath.Join(cwd, "../frontend/public")
	fmt.Println("Public File Path", fp)
	return fp
}

func GetTestDir() string {
	cwd, err := os.Getwd()
	if err != nil {
		panic(err)
	}
	fp := filepath.Join(cwd, "../tests")
	return fp
}

func ReadConfigBytes() ([]byte, error) {
	fePubDir := getFEPublicDir()
	fePubConfigPath := filepath.Join(fePubDir, "DBConfig.json")

	fePubConfigFile, err := ioutil.ReadFile(fePubConfigPath)
	if err != nil {
		return nil, err
	}
	return fePubConfigFile, nil
}
